app/src/org/commcare/CommCareApplication.java

```diff
@@ -301,22 +295,9 @@ public SecretKey createNewSymmetricKey() {
         return getSession().createNewSymmetricKey();
     }
 
-    synchronized public Tracker getDefaultTracker() {
-        if (analyticsTracker == null) {
-            if (BuildConfig.DEBUG) {
-                analyticsTracker = analyticsInstance.newTracker(DEV_TRACKING_ID);
-            } else {
-                analyticsTracker = analyticsInstance.newTracker(LIVE_TRACKING_ID);
-            }
-            analyticsTracker.enableAutoActivityTracking(true);
-        }
-        String userId = getCurrentUserId();
-        if (!"".equals(userId)) {
-            analyticsTracker.set("&uid", userId);
-        } else {
-            analyticsTracker.set("&uid", null);
-        }
-        return analyticsTracker;
+    synchronized public FirebaseAnalytics getAnalyticsInstance() {
+        analyticsInstance.setUserId(getUserIdOrNull());
+        return this.analyticsInstance;

```

Comment:
why `this`?

Reply:
no particular reason, i just use it sometimes. can remove